### PR TITLE
Fix exceptional case when loading a file in tsc_wrapped that has an `repo/external` segment in it

### PR DESCRIPTION
**Pull request by gregmagolan**
Not sure the best way to test for this. The failure is only in @angular//packages/bazel/test/ngc-wrapped:ngc_test which depends on `"//packages/bazel/src/ngc-wrapped:ngc_lib"` which depends on the typescript libs and then the test tries to load `.../ngc_test.sh.runfiles/angular/external/ngdeps/node_modules/typescript/lib/lib.d.ts` but this path is not valid when `--nolegacy_external_runfiles` is on.

`angular/external/ngdeps/node_modules/typescript/lib/lib.d.ts` is the correct path relative to the execroot but that path doesn't translate over nicely to runfiles as it used to when `--nolegacy_external_runfiles` is on.

@alexeagle Maybe you know of a better fix or workaround?

### CODE REVIEW

**Comment by gregmagolan** [comment]
CI build failing as https://github.com/bazelbuild/rules_typescript/pull/312 needs to land

File: internal/tsc_wrapped/file_cache.ts [hunk 1]
```diff
@@ -306,7 +306,23 @@ export class UncachedFileLoader implements FileLoader {
...
+    } catch (e) {
+      // If the file path is from an external repository then trim off the
+      // `repo/external` segments from the path so it matches the runfiles
+      // layout for when --nolegacy_external_runfiles flag is on and try again.
+      // This will modify a path such as
+      // `.../ngc_test.sh.runfiles/angular/external/ngdeps/node_modules/typescript/lib/lib.d.ts'`
+      // to `.../ngc_test.sh.runfiles/ngdeps/node_modules/typescript/lib/lib.d.ts'`.
+      // This fixes a test failure in @angular//packages/bazel/test/ngc-wrapped:ngc_test
+      // when --nolegacy_external_runfiles is turned on
+      if (filePath.match(/\.runfiles[\\\/][^\\\/]*[\\\/]external[\\\/]/)) {
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply internal/tsc_wrapped/file_cache.ts]: Good call. Thanks

**Comment by gregmagolan** [comment]
> Why don't we need the same fix in the CachedFileLoader implementation?

The issue is a corner case that only showed up when running the `@angular//packages/bazel/test/ngc-wrapped:ngc_test` which uses the UncachedFileLoader. tsc_wrapped goes with that loader when there are no inputs:

```
  if (inputs) {
    fileLoader = new CachedFileLoader(fileCache);
    ...
  } else {
    fileLoader = new UncachedFileLoader();
  }
```

The same issue would show up as well if there were inputs. I could make generalize the fix to handle both?

**Comment by gregmagolan** [comment]
Fixed root case in angular/angular so this work-around no longer necessary
